src/settingsview.cpp

```diff
@@ -119,22 +113,27 @@ void SettingsView::browseMonitorDir()
     if (dir == monitorDir || dir.isEmpty()) {
         return;
     }
-    if (confirmDialogMonitorDir(dir, false)) {
+    if (confirmDialogMonitorDir(dir)) {
         KiwixApp::instance()->getSettingsManager()->setMonitorDir(dir);
         ui->monitorClear->show();
-        KiwixApp::instance()->getLibrary()->loadMonitorDir(dir);
         KiwixApp::instance()->getWatcher()->addPath(dir);
+        auto previousDir = KiwixApp::instance()->getSettingsManager()->getMonitorDir();
+        if(previousDir != "") {
+            KiwixApp::instance()->getWatcher()->removePath(previousDir);
+        }
+        KiwixApp::instance()->getLibrary()->loadMonitorDir(dir);
```

Comment:
1. This call to thread-unsafe `loadMonitorDir()` (instead of the thread-safe `loadMonitorDirInThread()`) can become a source of random (though, probably, extremely rare) glitches (when it happens to be executed concurrently with a call of itself triggered by `QFileSystemWatcher::directoryChanged`).
2. `SettingsManager::m_monitorDirZims` must be cleared when changing the monitory directory, otherwise inside `loadMonitorDir()` called from here you will be comparing the contents of two different directories (which may have unexpected and hardly desirable effects)

Reply:
Thanks, missed that :}